skore/tests/unit/reports/estimator/feature_importance/test_permutation_importance.py

```diff
@@ -314,3 +315,99 @@ def test_not_fitted(regression_data):
     error_msg = "This LinearRegression instance is not fitted yet"
     with pytest.raises(NotFittedError, match=error_msg):
         report.feature_importance.permutation()
+
+
+class TestAtStep:
+    @pytest.fixture
+    def pipeline_report(self) -> EstimatorReport:
+        X, y = make_regression(n_features=3, random_state=0)
+
+        pipeline = make_pipeline(
+            StandardScaler(), PCA(n_components=2), LinearRegression()
+        )
+
+        return EstimatorReport(pipeline, X_train=X, y_train=y, X_test=X, y_test=y)
+
+    @pytest.mark.parametrize("at_step", [0, -1, 1])
+    def test_int(self, pipeline_report, at_step):
+        """
+        Test the `at_step` integer parameter for permutation importance with a pipeline.
+        """
+
+        result = pipeline_report.feature_importance.permutation(
+            seed=42, at_step=at_step
+        )
+
+        assert isinstance(result.index, pd.MultiIndex)
+        assert result.index.nlevels == 2
+        assert result.index.names == ["Metric", "Feature"]
+        assert result.shape[0] > 0
```

Comment:
Since we have a scikit-learn pipeline, let's check the "Features" from the dataframe as well. It should corresponds to the `feature_names_in_` of the steps